crates/ol/mempool/src/validation.rs

```diff
@@ -0,0 +1,204 @@
+//! Transaction validation for mempool.
+//!
+//! Provides pluggable validation strategies to check transaction validity before inclusion.
+
+use strata_db_types::types::MempoolTxMetadata;
+use strata_ol_chain_types_new::OLTransaction;
+
+use crate::{MempoolError, MempoolResult};
+
+/// Strategy for validating transactions before mempool inclusion.
+pub trait TransactionValidator: Send + Sync {
+    /// Validate a transaction for mempool inclusion.
+    ///
+    /// Returns Ok(()) if the transaction is valid, or an error describing the issue.
+    fn validate(
+        &self,
+        tx: &OLTransaction,
+        metadata: &MempoolTxMetadata,
+        current_slot: u64,
+    ) -> MempoolResult<()>;
+
+    /// Name of the validator (for logging/metrics).
+    fn name(&self) -> &'static str;
+}
+
+/// Basic validator that checks fundamental transaction properties.
+///
+/// Validates:
+/// - Transaction size within limits
+/// - min_slot <= max_slot (if both present)
+/// - Transaction not expired (max_slot >= current_slot)
+/// - Transaction not too early (min_slot > current_slot)
+#[derive(Debug, Clone, Copy)]
+pub struct BasicValidator {
+    /// Maximum transaction size in bytes.
+    pub max_tx_size: usize,
+}
+
+impl BasicValidator {
+    /// Create a new basic validator with the given size limit.
+    pub fn new(max_tx_size: usize) -> Self {
+        Self { max_tx_size }
+    }
+}
+
+impl TransactionValidator for BasicValidator {
+    fn validate(
+        &self,
+        tx: &OLTransaction,
+        metadata: &MempoolTxMetadata,
+        current_slot: u64,
+    ) -> MempoolResult<()> {
+        // Check transaction size
+        if metadata.size_bytes > self.max_tx_size {
+            return Err(MempoolError::TransactionTooLarge {
+                size: metadata.size_bytes,
+                max: self.max_tx_size,
+            });
+        }
+
+        // Validate slot ranges
+        let min_slot = tx.attachments().min_slot();
+        let max_slot = tx.attachments().max_slot();
+
+        // If both min_slot and max_slot are present, min must be <= max
+        if let (Some(min), Some(max)) = (min_slot, max_slot)
+            && min > max
+        {
+            return Err(MempoolError::InvalidTransaction(format!(
+                "Invalid slot range: min_slot ({min}) > max_slot ({max})",
+            )));
+        }
+
+        // Check if transaction is too early
+        if let Some(min) = min_slot
+            && min > current_slot
+        {
+            return Err(MempoolError::TooEarly {
+                min_slot: min,
+                current_slot,
+            });
+        }
+
+        // Check if transaction is expired
+        if let Some(max) = max_slot
+            && max < current_slot
+        {
+            return Err(MempoolError::Expired { max_slot: max });
+        }
```

Comment:
This doesn't actually validate the transaction, it just checks the tx attachment's preconditions.